### PR TITLE
bpo-35436: Add missing PyErr_NoMemory() calls and other minor bug fixes

### PR DESCRIPTION
**Pull request by ZackerySpytz**
Set MemoryError when appropriate, add missing failure checks,
and fix some potential leaks.

https://bugs.python.org/issue35436

### CODE REVIEW

File: Parser/myreadline.c [hunk 1]
```diff
@@ -211,17 +228,23 @@ PyOS_StdioReadline(FILE *sys_stdin, FILE *sys_stdout, const char *prompt)
...
-                            fflush(stderr);
-                            /* wlen includes null terminator, so subtract 1 */
-                            WriteConsoleW(hStdErr, wbuf, wlen - 1, &n, NULL);
+                    if (wlen) {
+                        wbuf = PyMem_RawMalloc(wlen * sizeof(wchar_t));
+                        if (wbuf == NULL) {
+                            PyErr_NoMemory();
+                            return NULL;
+                        }
+                        else {
```
Comment (serhiy-storchaka) [inline Parser/myreadline.c]: You can remove `else` and decrease an indentation level.

**Review by serhiy-storchaka — APPROVED** [review]
Great work! It is worth adding a news entry.

**Comment by ZackerySpytz** [comment]
@serhiy-storchaka Thank you for the review.

**Review by serhiy-storchaka — APPROVED** [review]

**Comment by ZackerySpytz** [comment]
@serhiy-storchaka I'll create the backport PRs.

**Pull request merged**